src/Symfony/Component/Validator/Tests/Constraints/FileTest.php

```diff
@@ -101,11 +133,17 @@ public static function provideValidSizes()
             ['3M', 3000000, false],
             ['1gi', 1073741824, true],
             ['1GI', 1073741824, true],
-            ['4g', 4000000000, false],
-            ['4G', 4000000000, false],
+            ['2g', 2000000000, false],
+            ['2G', 2000000000, false],
         ];
     }
 
+    public static function provideValidSizesForIntegerOverflow(): \Generator
+    {
+        yield ['4g', '4000000000', false];
+        yield ['4G', '4000000000', false];
```

Comment:
I think we should also have a test for 64 bit systems exclusively that ensure the value is an integer.

Reply:
Or maybe add a flag to the existing data provider which indicates if the provided fixture exceeds `PHP_INT_MAX` and cast to string on 32-bit systems?

Reply:
see #52263 for what I have in mind

Reply:
I think your way is cleaner and better!